### ISSUE TITLE
Mirror production environment for pentesting

### ISSUE THREAD

**Issue by BudickDa** [corona-warn-app/cwa-documentation#251]
## Environment that mirrors production

There will be a lot of people (including me) that will try to ~proof that you messed up~ improve the security of the system.

Running penetration test on a local deployment only tests the code. 
**But what about infrastructure, deployment pipeline, and configuration?**

Attacking the production environment is a big nono, but bad actors won't care.

With a highly sensitive topic like public health a successful attack could be literally fatal.
If one finds a way to artificially create infection or connections to infected people, a lot of good and important data could be polluted.
What about primitive attacks like DDoS? A DDoS should not take you offline. But what if it does?
Can I try it? The baddies won't ask, they will hammer you for month. Not now, not tomorrow (ok, maybe tomorrow at launch, there is always that guy), but when the next wave is here.

**tl;dr: 
Creating a mirror of the production environment would enable the good actors to research as effective as the bad ones.**

## Make fake data available
There should be a deployment that exactly mirrors the production environment, and it should be filled valid but fake data.
The data of the fake system is uploaded each day and is then reseted, so security researches can check if they could manipulate the data with fake requests.

Example:
Fake Data:
Person A had contact to Person B
Person A is infected

If I could mark Person B as infected or could create Person C, that had contact to Person A, it could be proof for a vulnerability. Maybe in the infrastructure, the process...

**Comment by tkowark** [comment]
Moved to documentation in case this needs to be expanded to other systems, too.

**Comment by haxxbard** [comment]
We will discuss this further; nevertheless you are (as you also wrote) able to run tests in your local environment with the information we provided.
